lib/pacemaker/pcmk_graph_consumer.c

```diff
@@ -557,13 +557,13 @@ unpack_action(pcmk__graph_synapse_t *parent, xmlNode *xml_action)
  * \internal
  * \brief Unpack transition graph synapse from XML
  *
- * \param[in] new_graph    Transition graph that synapse is part of
- * \param[in] xml_synapse  Synapse XML
+ * \param[in,out] new_graph    Transition graph that synapse is part of
+ * \param[in]     xml_synapse  Synapse XML
  *
  * \return Newly allocated synapse on success, or NULL otherwise
  */
 static pcmk__graph_synapse_t *
-unpack_synapse(pcmk__graph_t *new_graph, xmlNode *xml_synapse)
+unpack_synapse(pcmk__graph_t *new_graph, const xmlNode *xml_synapse)
```

Comment:
We should hit `unpack_action()` too and make the appropriate variables const in `unpack_synapse()` (i.e., the ones we access via `xml_synapse`).

Reply:
unpack_action()'s `parent` is `[in]` but not `const` because it's used to set a non-const struct member, and `xml_action` is also `[in]` but not `const` due to being used as an argument to a libxml2 function. And of course being used with unpack_action() is why the variables here aren't const.

Reply:
That makes sense, I didn't look closely enough. Having `xml_synapse` const here is a gray area since we're accessing other objects through it that we're then accessing as non-const. I lean toward "it's fine," since we're not modifying `xml_synapse` itself **and** the accessed objects are passed as [in] arguments.

Following that logic though, `action_set` in the outer for-loop could probably be const too, but that looks a bit more egregious.

If we ever go down the rabbit hole of sorting out which functions do or don't need to create a new doc pointer, maybe this ends up cleaned up. Until then, we'll keep sighing at XML const-ness.